framework/src/problems/FEProblemBase.C

```diff
@@ -5870,28 +5857,69 @@ FEProblemBase::checkExceptionAndStopSolve(bool print_message)
   {
     _communicator.broadcast(_exception_message, processor_id);
 
-    // Print the message
-    if (_communicator.rank() == 0 && print_message)
-      Moose::err << _exception_message << std::endl;
+    if (_current_execute_on_flag == EXEC_LINEAR || _current_execute_on_flag == EXEC_NONLINEAR ||
+        _current_execute_on_flag == EXEC_POSTCHECK)
+    {
+      // Print the message
+      if (_communicator.rank() == 0 && print_message)
+      {
+        _console << "\n" << _exception_message << "\n";
+        if (isTransient())
+          _console
+              << "To recover, the solution will fail and then be re-attempted with a reduced time "
+                 "step.\n"
+              << std::endl;
+      }
 
-    // Stop the solve -- this entails setting
-    // SNESSetFunctionDomainError() or directly inserting NaNs in the
-    // residual vector to let PETSc >= 3.6 return DIVERGED_NANORINF.
-    _current_nl_sys->stopSolve();
+      // Stop the solve -- this entails setting
+      // SNESSetFunctionDomainError() or directly inserting NaNs in the
+      // residual vector to let PETSc >= 3.6 return DIVERGED_NANORINF.
+      _current_nl_sys->stopSolve(_current_execute_on_flag);
 
-    // and close Aux system (we MUST do this here; see #11525)
-    _aux->solution().close();
+      // and close Aux system (we MUST do this here; see #11525)
+      _aux->solution().close();
 
-    // We've handled this exception, so we no longer have one.
-    _has_exception = false;
+      // We've handled this exception, so we no longer have one.
+      _has_exception = false;
 
-    // Force the next non-linear convergence check to fail (and all further residual evaluation to
-    // be skipped).
-    _fail_next_nonlinear_convergence_check = true;
+      // Force the next non-linear convergence check to fail (and all further residual evaluation
+      // to be skipped).
+      _fail_next_nonlinear_convergence_check = true;
 
-    // Repropagate the exception, so it can be caught at a higher level, typically
-    // this is NonlinearSystem::computeResidual().
-    throw MooseException(_exception_message);
+      // Repropagate the exception, so it can be caught at a higher level, typically
+      // this is NonlinearSystem::computeResidual().
+      throw MooseException(_exception_message);
+    }
+    else
+      mooseError("The following parallel-communicated exception was detected during " +
```

Comment:
@lindsayad What do you think about this message? Would you rather just call it a "MooseException" like we did before? Maybe a "parallel-communicated MooseException"? I kind of like including the parallel bit because it clarifies the fact that it has been communicated in parallel already.

Reply:
I like this message. Yes the parallel bit is good. But also I like that it *doesn't* say MooseException since we handle more exception types than just MooseExceptions

Reply:
OK, great -- let's leave it as is.